nntrainer/tensor/blas_interface.cpp

```diff
@@ -159,6 +159,42 @@ static void scopy_FP16(const unsigned int N, const _FP16 *X, const int incX,
 #endif
 }
 
+static void scopy_float32_to_float16(const unsigned int N, const float *X,
+                                     const int incX, _FP16 *Y, const int incY) {
+  unsigned int incy = abs(incY);
+  unsigned int incx = abs(incX);
+
+#if (defined USE__FP16 && USE_NEON)
+  if (incX == 1 && incY == 1) {
+    nntrainer::neon::scopy_neon_fp32_to_fp16(N, X, Y);
+  } else {
+    for (unsigned int i = 0; i < N; ++i)
+      Y[i * incy] = X[i * incx];
```

Comment:
same here

Reply:
When using android __fp16, we do not really have to cast float to __fp16 and vice versa as far as I recognize.
On the contrary, when using _Float16, we cannot transform values to lower/higher precision without casting it.

Reply:
thanks for the clarification. so what u mean by "do not really have to cast" is  `__fp16` automatically transforms its value? (like how `float x = 2.23; int y = x;` works)

Reply:
That is correct.
For example, when you run something like:
```C++
  float v1 = 3.1;
  __fp16 v1_16 = v1;
  __fp16 v1_16_casted = static_cast<__fp16>(v1);
  float v1_32 = v1_16;
  float v1_32_casted = static_cast<float>(v1_16);

  float v2 = 3.14159262457; // __fp16 overflow
  __fp16 v2_16 = v2;
  __fp16 v2_16_casted = static_cast<__fp16>(v2);
  float v2_32 = v2_16;
  float v2_32_casted = static_cast<float>(v2_16);


  std::cout << "v1_16 : " << v1_16 << std::endl;
  std::cout << "v1_16_casted : " << v1_16_casted << std::endl;
  std::cout << "v1_32 : " << v1_32 << std::endl;
  std::cout << "v1_32_casted : " << v1_32_casted << std::endl;

  std::cout << "v2_16 : " << v2_16 << std::endl;
  std::cout << "v2_16_casted : " << v2_16_casted << std::endl;
  std::cout << "v2_32 : " << v2_32 << std::endl;
  std::cout << "v2_32_casted : " << v2_32_casted << std::endl;
```

Then you get:
```bash
v1_16 : 3.09961
v1_16_casted : 3.09961
v1_32 : 3.09961
v1_32_casted : 3.09961
v2_16 : 3.14062
v2_16_casted : 3.14062
v2_32 : 3.14062
v2_32_casted : 3.14062
```

Reply:
So technically, there is no use of `static_cast<__fp16>` in this specific situation.